botorch/acquisition/multi_objective/utils.py

```diff
@@ -161,3 +174,224 @@ def prune_inferior_points_multi_objective(
+        # `num_pareto_samples x 2 x (J - 1) x (M + K)`
+        hypercell_bounds = torch.cat([hypercell_bounds, feasible_boxes], dim=-1)
+
+        # `num_pareto_samples x 2 x 1 x (M + K)`
+        infeasible_box = torch.zeros(
+            hypercell_bounds.shape[:-2] + torch.Size([1, M + K]), **tkwargs
+        )
+        infeasible_box[..., 1, :, M:] = -NEG_INF
+        infeasible_box[..., 0, :, 0:M] = NEG_INF
+        infeasible_box[..., 1, :, 0:M] = -NEG_INF
+
+        # `num_pareto_samples x 2 x J x (M + K)`
+        hypercell_bounds = torch.cat([hypercell_bounds, infeasible_box], dim=-2)
+
+    # `num_pareto_samples x 2 x J x (M + K)`
+    return hypercell_bounds
+
+
+def random_search_optimizer(
+    model: GenericDeterministicModel,
+    bounds: Tensor,
+    num_points: int,
+    maximize: bool,
+    pop_size: int = 1024,
+    max_tries: int = 10,
+) -> Tuple[Tensor, Tensor]:
+    r"""Optimize a function via random search.
+
+    Args:
+        model: The model.
+        bounds: A `2 x d`-dim Tensor containing the input bounds.
+        num_points: The number of optimal points to be outputted.
+        maximize: If true, we consider a maximization problem.
+        pop_size: The number of function evaluations per try.
+        max_tries: The maximum number of tries.
+
+    Returns:
+        A two-element tuple containing
+
+        - A `num_points x d`-dim Tensor containing the collection of optimal inputs.
+        - A `num_points x M`-dim Tensor containing the collection of optimal
+            objectives.
+    """
+    tkwargs = {"dtype": bounds.dtype, "device": bounds.device}
+    weight = 1.0 if maximize else -1.0
+    optimal_inputs = torch.tensor([], **tkwargs)
+    optimal_outputs = torch.tensor([], **tkwargs)
+    num_tries = 0
+    ratio = 2
+    while ratio > 1 and num_tries < max_tries:
+        X = draw_sobol_samples(bounds=bounds, n=pop_size, q=1).squeeze(-2)
+        Y = model.posterior(X).mean
+        X_aug = torch.cat([optimal_inputs, X], dim=0)
+        Y_aug = torch.cat([optimal_outputs, Y], dim=0)
+        pareto_mask = is_non_dominated(weight * Y_aug)
+        optimal_inputs = X_aug[pareto_mask]
+        optimal_outputs = Y_aug[pareto_mask]
+        num_found = len(optimal_inputs)
+        ratio = ceil(num_points / num_found)
+        num_tries = num_tries + 1
+    # If maximum number of retries exceeded throw out a runtime error.
+    if ratio > 1:
+        error_text = f"Only found {num_found} optimal points instead of {num_points}."
+        raise RuntimeError(error_text)
+    else:
+        return optimal_inputs[:num_points], optimal_outputs[:num_points]
+
+
+def sample_optimal_points(
+    model: Model,
+    bounds: Tensor,
+    num_samples: int,
+    num_points: int,
+    optimizer: Callable[
+        [GenericDeterministicModel, Tensor, int, bool, Any], Tuple[Tensor, Tensor]
+    ] = random_search_optimizer,
+    num_rff_features: int = 512,
+    maximize: bool = True,
+    optimizer_kwargs: dict = None,
```

Comment:
```suggestion
    optimizer_kwargs: Optional[Dict[str, Any]] = None,
```
addressing a complaint from a Meta internal linter